tools/enrollment_preprocessor.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python3
+import csv
+from argparse import ArgumentParser
+from dataclasses import dataclass
+from io import BytesIO
+from typing import TextIO
+
+from openpyxl import load_workbook
+from openpyxl.cell import Cell
+
+
+@dataclass
+class User:
+    title: str
+    last_name: str
+    first_name: str
+    email: str
+
+
+@dataclass
+class UserCells:
+    title: Cell | None
+    last_name: Cell
+    first_name: Cell
+    email: Cell
+
+    def value(self) -> User:
+        return User(
+            str(self.title.value) if self.title and self.title.value else "",
+            str(self.last_name.value),
+            str(self.first_name.value),
+            str(self.email.value),
+        )
+
+
+def user_decision(field: str, existing: str, imported: str) -> str:
+    if existing == imported:
+        return existing
+    decision: str = ""
+    while decision not in ("y", "n"):
+        decision = input(f"Do you want to keep the existing user {field}? (y/n) ")
+    if decision == "n":
+        return imported
+    return existing
+
+
+def fix_user(users: dict[str, User], imported_cells: UserCells) -> None:
+    imported = imported_cells.value()
+    existing = users.setdefault(imported.email, imported)
+    if imported == existing:
+        return
+    print("There is a conflict in the user data.")
+    print(f"existing: {existing}.")
+    print(f"imported: {imported}.")
+    # None is passed exclusively for participants since they have no title column
+    if imported_cells.title is not None:
+        imported_cells.title.value = user_decision("title", existing.title, imported.title)
+    imported_cells.last_name.value = user_decision("last name", existing.last_name, imported.last_name)
+    imported_cells.first_name.value = user_decision("first name", existing.first_name, imported.first_name)
+    imported_cells.email.value = user_decision("email", existing.email, imported.email)
+    print()
+
+
+def run_preprocessor(enrollment_data: str | BytesIO, user_data: TextIO) -> BytesIO:
+    workbook = load_workbook(enrollment_data)
+    users = {}
+    reader = csv.reader(user_data, delimiter=";", lineterminator="\n")
+    for row in reader:
+        email = row[-1]
+        users[email] = User(*row)
```

Comment:
Maybe, to not duplicate the "last column is email" knowledge:
```suggestion
        user = User(*row)
        users[user.email] = user
```